msg-socket/src/req/driver.rs

```diff
@@ -57,14 +59,16 @@ impl<T: AsyncRead + AsyncWrite> ReqDriver<T> {
     }
 
     fn on_message(&mut self, msg: reqrep::Message) {
-        if let Some(pending) = self.pending_requests.remove(&msg.id()) {
+        if let Some(pending) = self.pending_requests.remove(&msg.id()) {     
             let rtt = pending.start.elapsed().as_micros() as usize;
             let size = msg.size();
             let _ = pending.sender.send(Ok(msg.into_payload()));
 
             // Update stats
             self.socket_state.stats.update_rtt(rtt);
             self.socket_state.stats.increment_rx(size);
+
+            self.active_requests.fetch_sub(1, Ordering::SeqCst);
```

Comment:
We can probably use `Ordering::Relaxed` everywhere since we're dealing with a "soft limit". This memory ordering is more performant.